pkg/amqp/topology_builder.go

```diff
@@ -11,11 +11,10 @@ import (
 // Default TopologyBuilder is DefaultTopologyBuilder.
 // If you need custom built topology, you should implement your own TopologyBuilder and pass it to the amqp.Config:
 //
-// 	config := NewDurablePubSubConfig()
-// 	config.TopologyBuilder = MyProCustomBuilder{}
-//
+//	config := NewDurablePubSubConfig()
+//	config.TopologyBuilder = MyProCustomBuilder{}
 type TopologyBuilder interface {
-	BuildTopology(channel *amqp.Channel, queueName string, exchangeName string, config Config, logger watermill.LoggerAdapter) error
+	BuildTopology(channel *amqp.Channel, queueName string, exchangeName string, routingKey string, config Config, logger watermill.LoggerAdapter) error
```

Comment:
This changes the public interface, so we would need to bump the major version to do this.

Reply:
That is absolutely correct. Do you have a suggestion on how to solve the routing problem without breaking change? Otherwise, the only solution would be to create your own TopologyBuilder and set an internal variable before each BuildTopology call.

On the other hand, is the handling of routing within the BuildTopology function not so wrong?

For me at least, the PR would no longer be a priority because we have switched to another broker, but the problem seems to exist due to the requests from other developers

Reply:
I think we'll need to bump major with this. I'll see if we can have some more extendable API here, like passing a struct instead of positional arguments.